lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -430,11 +430,30 @@ public function processParameterValue($value)
             // Silence any mapping exceptions. These can occur if the object in
             // question is not a mapped entity, in which case we just don't do
             // any preparation on the value.
+
+            // If no mapping is detected, trying to resolve the value as a Traversable
+            if ($value instanceof Traversable) {
+                $value = iterator_to_array($value);
+                $this->processArrayParameterValue($value);
+            }
         }
 
         return $value;
     }
 
+    /**
+     * Process a parameter value which was previously identified as an array
+     *
+     * @param mixed[] $value
+     */
+    private function processArrayParameterValue(array &$value) : void
```

Comment:
Can you please pass this argument by value instead of by reference and return a value?

Reply:
Of course... Updated :)